client/src/redux/project/projectSlice.ts

```diff
@@ -0,0 +1,529 @@
+import {
+  createSlice,
+  createAsyncThunk,
+  ActionReducerMapBuilder,
+  PayloadAction
+} from '@reduxjs/toolkit'
+
+import { Project, AxiosResponseError } from '~/shared/types'
+import { catchError } from '~/utils/handleAxiosError'
+import projectService from './projectService'
+
+type InitialState = {
+  project: Project | null
+  sidebarProject: any
+  overviewProject: Project | null
+  newProject: {
+    title: string,
+    description: string,
+    teams: any
+  }
+  renameTeamData: {
+    projectID: number,
+    teamID: number,
+    name: string
+  }
+  addNewTeam: {
+    projectID: number,
+    teamData: any
+  }
+  removeTeam: {
+    projectID: number,
+    teamID: number
+  }
+  projectDescription: {
+    id: number,
+    title: string,
+    description: string
+  }
+  refresher: {
+    teamStateUpdate: boolean,
+    memberStateUpdate: boolean,
+    projectStateUpdate: boolean,
+  }
+  update: {
+    status: number,
+    projectID: number,
+  }
+  filter: number
+  isError: boolean
+  isSuccess: boolean
+  isLoading: boolean
+  error: AxiosResponseError
+};
+
+const initialState: InitialState = {
```

Comment:
You can use `Generic` types so that I would shorten the types but anyway I will approved it!